### ISSUE TITLE
Vtreat newvars Doesn't Take `make.names` into account

### ISSUE THREAD

**Issue by jcpetkovich** [WinVector/vtreat#23]
When giving variables to vtreat that are not valid R variable names, the `vnames` function no longer returns a translation from the original non-valid R variable names to the new valid R variable names.

As an example, if I rename the first column of the iris dataframe to: "mis be hav (%)"

And use vtreat on it like so:

```R
p <- designTreatmentsZ(iris, names(iris))
```

When I use prepare, I get the following:

```R
> names(prepare(p, iris))[1]
[1] "mis.be.hav...."
```
When I inspect the vorig and vnames of this treatment, I see the following:

```R
> vorig(p$treatments[[1]])
[1] "mis be hav (%)"
> vnames(p$treatments[[1]])
[1] "mis be hav (%)"
```

When what I expect to see is:

```R
> vorig(p$treatments[[1]])
[1] "mis be hav (%)"
> vnames(p$treatments[[1]])
[1] "mis.be.hav...."
```

**Comment by JohnMount** [comment]
Jean-Christophe Petkovich, sorry you ran into trouble and thank you for investigating it and turning in a clear issue report.

It looks like we introduced this problem on our side in vtreat 1.3.3 2018/12/17 when we removed the processing that added the "_clean" suffix to (NA-fixed) passed-through variables.  Nina Zumel and I are thinking how we want to handle this and will hopefully get a fix in shortly.

**Comment by JohnMount** [comment]
I have fixed this issue in the development version.  Note the name-munging is a bit different since you last saw it (it prefers "_" to dot). I have also entered [a test which captures the effect you reported](https://github.com/WinVector/vtreat/blob/master/inst/unit_tests/test_name_munging.R).

Again, thank you for using vtreat, sorry for the trouble, and thank you for taking the time to submit an issue.
